### PR TITLE
Report and debug permanent 403s (or corrupt M3U8)

### PR DESCRIPTION
**Pull request by tchakabam**
@pokey909 @ohookins this should be doing it.

### CODE REVIEW

**Comment by ohookins** [comment]
No idea :)

File: src/SkippyM3UParser.cpp [hunk 1]
```diff
@@ -80,6 +80,8 @@ SkippyM3UPlaylist SkippyM3UParser::parse(string uri, const string& playlist)
   // Output playlist
   SkippyM3UPlaylist outputPlaylist(uri);
 
+  g_message ("Dumping whole M3U8:\n\n\n%s\n\n\n", playlist.c_str());
```
Comment (tchakabam) [inline src/SkippyM3UParser.cpp]: This is going to dump the whole M3U8 as it has been loaded before we parse it.

It's actually not necessary to log all the parsing process. If the M3U8 is fine before this, then we know the error is in the parsing. If its not fine here, then it must have happened on the transport or in the HTTP client.

Comment (ohookins) [inline src/SkippyM3UParser.cpp]: Are we going to have enough space in the logging ring buffer? Some playlists can be hundreds of KB to a couple of MB.

Comment (tchakabam) [inline src/SkippyM3UParser.cpp]: I could see that a 2 hour track was about ~900 lines (will check on the MB but that should be very linear with the duration of the track). That will definitely take a lot of space and that means we might not see the whole thing in the logs the users send. We might rather attach it to the error report itself. It's probably OK for CrashLytics if we do this only when it happens (I don't think it matters to them if we just report an error or upload some data with it, it's the same amount of requests to process).

**Comment by pokey909** [comment]
Considering we have 5000 line limit set per crashlytics event, that should be enough to see if some manifests are broken. :+1:

**Comment by tchakabam** [comment]
@pokey909 i suspect that it gets truncated somewhere because the reports are usually less than 5000 lines long :( but lets hope we catch it somehow ;)

**Pull request merged**